### DISCUSSION TITLE
Unable to connect via TCP with a friend to play

### DISCUSSION THREAD

**Discussion by Frankie4daKill** [diasurgical/DevilutionX#1649] [General]
Hello
me and my friend tried today to connect via TCP to play together. for some reason even if we try each other IPs, it does not let us play together as we getting message of either "unable to connect" or as if the request expired because the other side didnt accept, when in fact he couldnt see any request from me.
we are using Windows 10
please help

**Comment by AJenbo** [comment]
You need to forwarded TCP port 6112 if you have a router.

Test where the problem is by running two instances of the game on your own computer.
Start with 127.0.0.1, then your lan IP and then internet IP

You can also try the latest beta which has an option for using ZeroTier, that way you don't need to worry about port forwarding at all.
https://app.circleci.com/pipelines/github/diasurgical/devilutionX?branch=master
(game name will be in the map once created)

**Reply by Frankie4daKill** [reply]
first of all, thank you for replying.
second, me and my friend downloaded and installed the ZeroTier, were able to connect to each other, but once we open the map the name is stil 0.0.0.0  and we cannot change it, and we cannot connect that IP of course it doesnt work.
if you can please explain how to make that we see the correct address after we open the map itself
thank you

**Reply by AJenbo** [reply]
`0.0.0.0` is the bound IP when hosting via TCP, if you see it you are _not_ using ZeroTier for hosting the game. For games hosted via ZeroTier you will see something like `3jwi5sq`.

It is also not your IP, but the network binding. `0.0.0.0` means that it's not bound and is available on all your connected networks.
To figure out what your actual internet IP is you need to look it up in your router or visit a website that will tell you what IP it sees you coming from, like https://www.myip.com/
We cannot auto-detect your public internet IP for you as it is not visible to the computer itself.

**Reply by Frankie4daKill** [reply]
We tried through checking our IP online of course, it didnt work. just not responding.
how can i host my game through ZeroTier? even when our connection status in ZeroTier is OK, the game address remain 0.0.0.0 thats the problem, its like the fact that we are connected in ZeroTier does nothing

**Reply by AJenbo** [reply]
Again use the build ZeroTier feature in the latest beta: https://app.circleci.com/pipelines/github/diasurgical/devilutionX?branch=master

You are clearly doing something else. If you pick TCP in-game you are doing it wrong.

**Reply by Frankie4daKill** [reply]
yeah we are playing the DevilutionX
wait what, "If you pick TCP in-game you are doing it wrong."  should i not pick TCP in game? what to pick then?
(sorry im so slow to understand, trying my best honestly)

**Reply by AJenbo** [reply]
have you ddownloaded the beta that i keep linking?

you should click zerotier not tcp

**Reply by Frankie4daKill** [reply]
clicking ZeroTier INSIDE the devolutionX ? :O
i didnt see such option inside the game, even when we were connected via ZeroTier

**Reply by Frankie4daKill** [reply]
brother do you have discord? maybe i can share screen and you explain me how please? =]

**Reply by AJenbo** [reply]
> clicking ZeroTier INSIDE the devolutionX ? :O
> i didnt see such option inside the game, even when we were connected via ZeroTier

download the beta: https://app.circleci.com/pipelines/github/diasurgical/devilutionX?branch=master please

you should not set up ZeroTier manually

discored link is at the top of our readme
